File: x8086NetEmuRenderers/Extensions.vb
Imports System.Runtime.CompilerServices
Imports x8086NetEmu.Adapter

Module Extensions

    <Extension()>
    Public Function ToPoint(v As XPoint) As Point
        Return New Point(v.X, v.Y)
    End Function

    <Extension()>
    Public Function ToSize(v As XSize) As Size
        Return New Size(v.Width, v.Height)
    End Function

    <Extension()>
    Public Function ToRectangle(v As XRectangle) As Rectangle
        Return New Rectangle(v.X, v.Y, v.Width, v.Height)
    End Function

    <Extension()>
    Public Function ToColor(v As XColor) As Color
        'Return Color.FromArgb(v.ToArgb())
        Return Color.FromArgb(v.A, v.R, v.G, v.B)
    End Function
End Module

File: Eto.Forms/x8086NetEmuEto/Extensions.cs
using Eto.Drawing;
using static x8086NetEmu.Adapter;

namespace x8086NetEmuEto {
    public static class Extensions {
        public static Point ToPoint(this XPoint v) {
            return new Point(v.X, v.Y);
        }

        public static Size ToSize(this XSize v) {
            return new Size(v.Width, v.Height);
        }

        public static Rectangle ToRectangle(this XRectangle v) {
            return new Rectangle(v.X, v.Y, v.Width, v.Height);
        }

        public static Color ToColor(this XColor v) {
            return Color.FromArgb(v.R, v.G, v.B, v.A);
        }
    }
}
